extensions/wikia/Wall/WallThread.class.php

```diff
@@ -77,31 +81,31 @@ private function loadReplyObjs() {
 		}
 	}
 
-	private function loadReplyIdsFromDB($master = false) {
+	/**
+	 * Fetches reply IDs for the thread, using a limit to control large queries
+	 *
+	 * @param DatabaseBase $dbr Database resource
+	 * @param integer $afterId The last reply ID after which the next set is selected
+	 * @return array List of reply IDs
+	 */
+	private function getReplyIdsFromDB( $dbr, $afterId = null ) {
 		// this is a direct way to get IDs
 		// the other one is in Wall.class done in a grouped way
 		// (fetch for many threads at once, set with ->setReplies)
 
-		$title = Title::newFromId( $this->mThreadId );
+		$conditions = array( 'parent_comment_id = '.$this->mThreadId );
 
-		if( empty($title) ) {
-			$title = Title::newFromId( $this->mThreadId, Title::GAID_FOR_UPDATE );
+		if ( (int) $afterId > 0 ) {
+			array_push( $conditions, 'comment_id > '.$afterId );
 		}
 
-		if( empty($title) ) {
-			return ;
-		}
-
-		$dbr = wfGetDB( $master ? DB_MASTER : DB_SLAVE );
-
-		$threadId = $title->getArticleID();
-
 		$result = $dbr->select(
 				array( 'comments_index' ),
 				array( 'distinct comment_id' ),
-				array( 'parent_comment_id = '.$threadId ),
+				$conditions,
 				__METHOD__,
-				array( 'ORDER BY' => 'comment_id ASC' )
+				array( 'ORDER BY' => 'comment_id ASC',
+					'LIMIT' => self::DB_LIMIT_REPLIES )
```

Comment:
I'm not familiar with FluentSQL, and it would take me a couple hours to learn it. And while that's worthwhile, I don't think it's worthwhile for this particular change. Unless you or another reviewer strongly disagree.